subprojects/plugins/src/main/java/org/gradle/jvm/component/JvmSoftwareComponent.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.component;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.component.SoftwareComponent;
+
+/**
+ * Base {@link SoftwareComponent} for the JVM platform.
```

Comment:
IMO, the core definition of a software component is the fact that it contains variants. However, there are some technical roadblocks to adding a `getVariants` method directly on `SoftwareComponent`. Particularly due to `ComponentWithVariants` already defining a `getVariants` method, but with the wrong return type. 

As for this interface itself, we can get rid of it for this PR. Eventually we will want a well defined interface for a JVM component, which may include accessing the runtime/compile classpath, runtime/compile elements, jar task, compile task, etc. All of this belongs on the component's interface, but I wanted to avoid adding it in this PR so that we can properly spec out and agree on what a component should be advertising as its interface. 

Where things get weird is multi-target components. There are several compile/runtime classpaths, jar tasks, compile tasks, java versions, etc. which makes it difficult to conceptualize what shared properties exist for any given JVM component. 

Conceptually, a single target and multi-target JVM component have separate interfaces. While a single-target can be modeled as a multi-target with only one target, this can be cumbersome to interact with from a DSL/buildscript standpoint. 

I have some ideas about how so solve this. But, perhaps its best to right now just remove the `JvmSoftwareComponent` interface from this PR until we better agree on what the interface should look like.

Reply:
Deleted `JvmSoftwareComponent` for now.